src/server/StudioManagementSystem.Infrastructure/Interfaces/Data/IStudentContactRepository.cs

```diff
@@ -5,5 +5,7 @@ namespace StudioManagementSystem.Infrastructure.Interfaces.Data;
 public interface IStudentContactRepository
 {
     Task<List<StudentContact>> GetAllStudentsAsync(CancellationToken ct);
+    Task<List<StudentContact>> GetAllStudentsWithNoGroupsAsync(CancellationToken ct);
+    Task<Guid> LeaveAssignedGroup(Guid id, CancellationToken ct);
```

Comment:
Likewise, this is the responsibility of the application layer (not data-layer repo's). This could be part of the same `GroupManager` that the rejection code lives in.